server/src/test/java/org/elasticsearch/action/admin/indices/refresh/RefreshResponseTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.refresh;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.support.DefaultShardOperationFailedException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.test.XContentTestUtils.insertRandomFields;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class RefreshResponseTests extends ESTestCase {
+
+    private static List<DefaultShardOperationFailedException> failures = new ArrayList<>();
+
+    private static class FakeElasticsearchException extends ElasticsearchException {
+
+        private Index index;
+        private ShardId shardId;
+        private RestStatus status;
+
+        public FakeElasticsearchException(String index, int shardId, RestStatus status, String msg) {
```

Comment:
do we need to subclass ElasticsearchException here? Can't we use ElasticsearchException directly?

Reply:
Yeah we can use `ElasticsearchException` directly, I did this at the beginning just because I want to set an `index` and `shardId` for an `ElasticsearchException`. If we create it directly it will be more complicated to set them ? Otherwise we can leave it `null`, like what I do now. Seems it's also okay here and not a big problem ?